subsys/net/lib/dns/dns_sd.c

```diff
@@ -822,6 +815,94 @@ int dns_sd_handle_ptr_query(const struct dns_sd_rec *inst,
 	return offset;
 }
 
+int dns_sd_handle_service_type_enum(const struct dns_sd_rec *inst,
+				    const struct in_addr *addr4, const struct in6_addr *addr6,
+				    uint8_t *buf, uint16_t buf_size)
+{
+	static const char query[] = { "\x09_services\x07_dns-sd\x04_udp\x05local" };
+	/* offset of '.local' in the above */
+	uint16_t domain_offset = DNS_SD_PTR_MASK | 35;
+	uint16_t proto;
+	int name_size;
+	uint16_t service_size;
+	uint16_t offset = sizeof(struct dns_header);
+	struct dns_rr *rr;
+	struct dns_header *rsp = (struct dns_header *)buf;
+
+	memset(rsp, 0, sizeof(*rsp));
+
+	if (!rec_is_valid(inst)) {
+		return -EINVAL;
+	}
+
+	if (*(inst->port) == 0) {
+		NET_DBG("Ephemeral port %u for %s.%s.%s.%s "
+			"not initialized",
+			ntohs(*(inst->port)), inst->instance, inst->service, inst->proto,
```

Comment:
The `ntohs()` looks unnecessary as the value is 0, so you can replace it by literal `0` value

Reply:
This matches what is done for PTR handling, so if it were changed here it would need to be changed there as well. I would prefer to address this in the larger rework task for v3